code/modules/urist/antagonist/revenant/revenant_slaps.dm

```diff
@@ -0,0 +1,253 @@
+
+	var/list/applicable_slaps = list()
+
+	for(var/datum/power/revenant/bs_slap/cand_slap in src.slaps)
+		if(cand_slap.distortion_threshold && effective_distortion > cand_slap.distortion_threshold)
+			applicable_slaps.Add(cand_slap)
+
+	if(!(applicable_slaps.len))
+		return
+
+	var/datum/power/revenant/bs_slap/slap = pick(applicable_slaps)
+
+	if(prob(99))
+		to_chat(M, SPAN_WARNING(slap.warning_message))
+		return FALSE
+
+	to_chat(M, SPAN_DANGER(slap.slap_message))
+
+	. = slap.Slap(M)
+	return
+
+
+/* Gibs the BSR mob. */
+/datum/power/revenant/bs_slap/gib
+	name = "Gibs"
+	flavor_tags = list(
+		BSR_FLAVOR_CHIMERA,
+		BSR_FLAVOR_VAMPIRE,
+		BSR_FLAVOR_DEMONIC,
+		BSR_FLAVOR_BLUESPACE,
+		BSR_FLAVOR_DENTIST,
+		BSR_FLAVOR_FLESH,
+		BSR_FLAVOR_GENERIC
+	)
+	warning_message = "You can feel your organs twitching and pulsing erratically! You'd better satisfy your Hunger fast!"
+	slap_message = "You have neglected your Hunger for too long! You feel your tendons snapping and your muscles tearing away from the bone as your organs violently burst..."
+
+
+/datum/power/revenant/bs_slap/gib/Slap(var/mob/M)
+	if(!istype(M))
+		return FALSE
+
+	M.gib()
+	return TRUE
+
+
+/* Explodes the BSR mob. */
+/datum/power/revenant/bs_slap/boom
+	name = "Boom!"
+	flavor_tags = list(
+		BSR_FLAVOR_BLUESPACE,
+		BSR_FLAVOR_DENTIST,
+		BSR_FLAVOR_SPACE,
+		BSR_FLAVOR_CHAOTIC,
+		BSR_FLAVOR_SCIFI
+	)
+	warning_message = "You feel a quiet but intensifying buzz as your body begins to heat up and vibrate! You'd better satisfy your Hunger fast!"
+	slap_message = "You have neglected your Hunger for too long! You feel every molecule in your body vibrate faster and faster, achieving impossible speeds as you try to keep yourself together..."
+
+
+/datum/power/revenant/bs_slap/boom/Slap(var/mob/M)
+	if(!istype(M))
+		return FALSE
+
+	var/turf/T = get_turf(M)
+	if(!istype(T))
+		return FALSE
+
+	M.gib()
+	explosion(T, 2, 4, 8, 4)
+	return TRUE
+
+
+/* Slowly fade out of existence */
+/datum/power/revenant/bs_slap/fadeout
+	name = "Fadeout"
+	flavor_tags = list(
+		BSR_FLAVOR_CHAOTIC,
+		BSR_FLAVOR_SCIFI,
+		BSR_FLAVOR_DARK,
+		BSR_FLAVOR_DEATH,
+		BSR_FLAVOR_GENERIC
+	)
+	warning_message = "You notice your hands becoming slightly transparent as you lose your connection to reality! You'd better satisfy your Hunger fast!"
+	slap_message = "You have neglected your Hunger for too long! You no longer have the strength to fight against reality itself - you let go and start slowly drifting away from existence..."
+
+
+/datum/power/revenant/bs_slap/fadeout/Slap(var/mob/M)
+	if(!istype(M))
+		return FALSE
+
+	var/const/fadetime = 600 // 1 min
+
+	M.remove_bsrevenant_powers()
+
+	var/datum/bluespace_revenant/bsr = M?.mind?.bluespace_revenant
+	if(istype(bsr))
+		bsr.stop_ticker()
+		M.mind.bluespace_revenant = null
+
+	M.visible_message(SPAN_WARNING("\The [src] starts to fade out of existence!."))
+
+	animate(M, alpha=0, time=fadetime)
+
+	spawn(fadetime)
```

Comment:
```suggestion
	addtimer(CALLBACK(null, /proc/qdel, M), fadetime)
```

Reply:
I  H A T E  Bay's API for callbacks and I know for a fact that this will have to be maintained come next merge, but fiiiiiine 👀